### PR TITLE
Add wildcard support for scp

### PR DESCRIPTION
**Pull request by bingbing8**
Add wildcard support for scp
https://github.com/PowerShell/Win32-OpenSSH/issues/904
https://github.com/PowerShell/Win32-OpenSSH/issues/988

### CODE REVIEW

File: scp.c [hunk 1]
```diff
@@ -488,9 +489,44 @@ main(int argc, char **argv)
 	msetlocale();
 
 	/* Copy argv, because we modify it */
-	newargv = xcalloc(MAXIMUM(argc + 1, 1), sizeof(*newargv));
+	newargv = xcalloc(MAXIMUM(argc + 1, 1), sizeof(*newargv));	
+#ifdef WINDOWS
```
Comment (bagajjal) [inline scp.c]: Please add comments why the ifdef block is required.
It will help when we go upstream.

File: scp.c [hunk 2]
```diff
@@ -488,9 +489,44 @@ main(int argc, char **argv)
...
 	/* Copy argv, because we modify it */
-	newargv = xcalloc(MAXIMUM(argc + 1, 1), sizeof(*newargv));
+	newargv = xcalloc(MAXIMUM(argc + 1, 1), sizeof(*newargv));	
+#ifdef WINDOWS	
+	{		
+		char *p, *argdup;
+		int i;		
+		glob_t g;
+		int expandargc = 0;
+
```
Comment (bagajjal) [inline scp.c]: delete extra line and initialize the variables above.

File: scp.c [hunk 3]
```diff
@@ -488,9 +489,44 @@ main(int argc, char **argv)
...
+			if (p = colon(argdup))
+				convertToForwardslash(p);
+			else
+				convertToForwardslash(argdup);
+			if (glob(argdup, GLOB_NOCHECK, NULL, &g)) {
+				if (expandargc > argc)
+					newargv = xreallocarray(newargv, expandargc + 1, sizeof(*newargv));
+				newargv[expandargc++] = xstrdup(argdup);
+			}
+			else {
```
Comment (bagajjal) [inline scp.c]: move to line 513

File: regress/pesterTests/SCP.Tests.ps1 [hunk 4]
```diff
@@ -92,8 +115,8 @@ Describe "Tests for scp command" -Tags "CI" {
         # for the first time, delete the existing log files.
         if ($OpenSSHTestInfo['DebugMode'])
```
Comment (bagajjal) [inline regress/pesterTests/SCP.Tests.ps1:118]: Since we are copying logs.. we can delete DebugMode all together?

**Review by bagajjal — CHANGES_REQUESTED** [review]
.

File: regress/pesterTests/SCP.Tests.ps1 (hunk 4 above)
Comment (bingbing8) [reply regress/pesterTests/SCP.Tests.ps1:118]: Actually we are not always copying log. Also when DebugMode is true, we save both sourcedir and destinationdir and have the log file name match tid, which make troubleshooting easier.

File: scp.c (hunk 3 above)
Comment (bingbing8) [reply scp.c]: fixed

File: scp.c (hunk 2 above)
Comment (bingbing8) [reply scp.c]: fixed

File: scp.c (hunk 1 above)
Comment (bingbing8) [reply scp.c]: added commments

File: regress/pesterTests/SCP.Tests.ps1 (hunk 4 above)
Comment (bagajjal) [reply regress/pesterTests/SCP.Tests.ps1:118]: can you do the similar change in other files like sftp tests.

File: scp.c (hunk 1 above)
Comment (bagajjal) [reply scp.c]: Also mention the directory separator conversion "\\" to "/"

**Review by bagajjal — APPROVED** [review]
please take of  2 new comments before submit.

File: scp.c (hunk 1 above)
Comment (bingbing8) [reply scp.c]: I did mentioned it in the code

File: regress/pesterTests/SCP.Tests.ps1 (hunk 4 above)
Comment (bingbing8) [reply regress/pesterTests/SCP.Tests.ps1:118]: updated sftp and cfginclude tests

**Pull request merged**